.github/workflows/build-firmware.yml

```
@@ -6,46 +6,70 @@ on:
+          fetch-depth: 0
+          persist-credentials: true
+
+      - name: Run board generator and commit changes (action)
+        uses: ./.github/actions/generate-boards
+        with:
+          commit: 'true'
+        # action will run the generator and push changes if there are any.
+
+      - name: Make matrix from Buildscripts/boards.json
+        id: make-matrix
+        run: |
```

actionlint (1.7.8)

shellcheck reported issue in this script: SC2086:info:2:23: Double quote to prevent globbing and word splitting

(shellcheck)